### ISSUE TITLE
Change intrinsic array to `array` class from builtin array type.

### ISSUE THREAD

**Issue by rhysd** [rhysd/Dachs#52]
Now Dachs uses `[T x N]` type for arrays (e.g. `[1, 2, 3]` employs `[3 x i64] [i64 1, i64 2, i64 3]`).  However, the builtin array type has a fixed length.  Dachs should have a variable length array.

To migrate array type from current fixed-length array to variable length array smoothly, at first I'll introduce `array` class and move current fixed-length array type to `static_array` type.

```
class array
    data : static_array
    size : uint

    init(@data)
        @size := @data.size
    end

    # ...
end
```

Then I'll introduce the `array` class to array literal and operator `[]`.  Because Dachs doesn't have an overload of operators, I'll add operator `[]` for `array` class specially.

After this issue is completed, I'll shift memory allocation from stack to heap with `malloc`.

**Comment by rhysd** [comment]
Now I'm working in progress for this issue in [wip/issue/52](https://github.com/rhysd/Dachs/tree/wip/issue/52) branch.

**Comment by rhysd** [comment]
#54 should be done before getting to work this issue because many patchwork for operators are needed to make array class work before #54 is done.  If #54 is implemented, all operators like `operator[]` of `array` class will work as built-in array and such patchworks will be no longer needed.

**Comment by rhysd** [comment]
#54 had been done.
Now is the time to advance this issue.
